_drafts/gdbundle-plugin-manager.md

```diff
@@ -0,0 +1,557 @@
+take long.
+
+By default, gdbundle:
+
+1. Searches for installed Python packages that begin with the name `gdbundle_`.
+2. Calls the `gdbundle_load()` function of each plugin module.
+3. In each plugin's `gdbundle_load()` function, the plugin should source,
+   import, and configure the GDB scripts that it has bundled inside (or from
+   another package by way of a dependency).
+
+And that is all.
+
+> gdbundle should work with LLDB out of the box, as all that is required is
+> Python and a way to run the setup procedure at launch. Please check out the
+> [gdbundle README.md](https://github.com/memfault/gdbundle) for more
+> information and examples.
+
+### Requirements
+
+gdbundle has a couple of requirements.
+
+#### Python-compatible GDB
+
+gdbundle requires a GDB compiled with Python. You can check if your GDB is
+configured correctly by the following quick test:
+
+```
+$ gdb
+(gdb) python-interactive
+>>> import sys; sys.version
+'3.6.7 | packaged by conda-forge'
+```
+
+If a Python version is printed, everything should work! If not, there are a few
+things to try:
+
+- Brew and Ubuntu's Apt both install a GDB with Python enabled.
+- Your distribution might also include a `*-gdb-py` version with Python enabled.
+- Clone [binutils-gdb](https://github.com/bminor/binutils-gdb) and try compiling
+  it with the configuration flag `--with-python`.
+
+#### Modification to `.gdbinit`
+
+Just like Vim plugin managers require the user to edit their `~/.vimrc`,
+gdbundle requires a snippet of code to be added to a loaded `.gdbinit` file.
+
+```
+# -- GDBUNDLE_EDITS_BEGIN
+python
+
+import os,subprocess,sys
+# Execute a Python using the user's shell and pull out the sys.path (for site-packages)
+paths = subprocess.check_output('python -c "import os,sys;print(os.linesep.join(sys.path).strip())"',shell=True).decode("utf-8").split()
+# Extend GDB's Python's search path
+sys.path.extend(paths)
+
+import gdbundle
+gdbundle.init()
+
+end
+# -- GDBUNDLE_EDITS_END
+```
+
+For more information about exactly what is happening above, feel free to read
+the short section in
+[Using PyPi Packages with GDB](https://interrupt.memfault.com/blog/using-pypi-packages-with-GDB#setting-syspath-within-gdbinit).
+
+The first block **is required** to be able to use Python packages installed in
+the activated virtual environment, Conda environment, or non-default Python
+installation.
+
+The second part, `gdbundle.init()`, is what is new and will load installed
+gdbundle plugins.
+
+> NOTE: The major (and ideally minor) version of Python linked with GDB should
+> match the version that is activated at the time of launch, whether that is a
+> virtual environment, Conda environment, `brew` Python installation, etc.
+> Mismatched major versions will cause packages and plugins to either break in
+> mysterious ways or not show up at all. More information can be found in the
+> [gdbundle README.md](https://github.com/memfault/gdbundle)
+
+## gdbundle Example Plugin
```

Comment:
```suggestion
## Creating a gdbundle Plugin
```

Reply:
Crap!  These changes get lost sometimes :(. Thanks for finding that.